code/lib/cli/src/generators/baseGenerator.ts

```diff
@@ -58,13 +58,30 @@ const getExternalFramework = (framework: string) =>
 
 const getFrameworkPackage = (framework: string, renderer: string, builder: string) => {
   const externalFramework = getExternalFramework(framework);
+  const storybookBuilder = builder?.replace(/^@storybook\/builder-/, '');
+  const storybookFramework = framework?.replace(/^@storybook\//, '');
```

Comment:
The fix really is in the `builder` part!
The framework part was just to be safe, in case we mistakenly pass a value in a baseGenerator like `@storybook/xyz`.